code/game/objects/items.dm

```diff
@@ -698,7 +698,7 @@
 /// Gives one of our item actions to a mob, when equipped to a certain slot
 /obj/item/proc/give_item_action(datum/action/action, mob/to_who, slot)
 	// Some items only give their actions buttons when in a specific slot.
-	if(!item_action_slot_check(slot, to_who))
+	if(!item_action_slot_check(slot, action, to_who))
```

Comment:
Doesn't this mess with all existing implementations of `item_action_slot_check`.